app/src/main/java/com/example/maintenancespace/utilities/CsvWriter.java

```diff
@@ -0,0 +1,44 @@
+package com.example.maintenancespace.utilities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.util.Log;
+
+import com.example.maintenancespace.models.events.MaintenanceEventModel;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class CsvWriter {
+    public final static int CSV_WRITE_REQUEST_CODE = 1;
+    public static void generateMaintenanceReports(Activity activity,ArrayList<MaintenanceEventModel> events) {
+        try {
+            // Create a new intent for creating a file
+            Intent intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            intent.putExtra(Intent.EXTRA_TITLE, "events.csv");
+            intent.setType("text/csv");
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+            // Create a holding file for the data
+            File eventsFile = new File(activity.getFilesDir(), "events.csv");
```

Comment:
The `getFilesDir` method is making use of that provider in the `provider_paths.xml` file.